code/datums/gamemode/powers/powers.dm

```diff
@@ -0,0 +1,369 @@
+/datum/power			
+	var/name = "Power"
+	var/desc = "Placeholder"
+	var/helptext = ""
+	var/granttext = ""
+	var/passive = FALSE     //is this a spell or a passive effect?
+	var/spellpath           //Path to a verb that contains the effects.
+	var/cost                //the cost of this power
+	var/datum/role/role 
+	var/obj/abstract/screen/movable/spell_master/spellmaster
+
+	var/store_in_memory = FALSE
+
+//basic proc for limiting powers in monkeys or whatever, override this with checks
+/datum/power/proc/can_use(var/mob/user)
+	return TRUE
+
+//adding the powers to the current ones, override this with passive effects
+/datum/power/proc/add_power(var/datum/role/R)
+	if (!istype(R) || !R)
+		return FALSE
+	if(locate(src) in R.current_powers)
```

Comment:
I suggested 
```
if(locate(type) in R.current_powers)
```
earlier, not `locate(src)`. I'm not sure the latter works correctly.
If you know you're working with the same instances you can just use
```
if(src in R.current_powers)
```
or the equivalent
```
if(R.current_powers.Find(src))
```